src/Components/ArticleList.jsx

```diff
@@ -7,40 +7,67 @@ const ArticleList= ()=> {
     const [articles, setArticles] = useState([])
     const [isLoading, setIsLoading]=useState(false)
     const [searchParams, setSearchParams] = useSearchParams()
+    const [sortBy, setSortBy] = useState("created_at");
+    const [sortOrder, setSortOrder] = useState("DESC");
     
-        useEffect(() => {
-            setIsLoading(true);
-            fetchData(`/articles?${searchParams.toString()}`).then(({articles})=>{
-                setArticles(articles);
-                setIsLoading(false)
-            }).catch((err)=>{
-                console.log(err);
-            })
-        },[searchParams])
-        
 
-        if(isLoading){
-            return (
-                <p id='loading'>Fetching your articles ...</p>
-                )
-        }else{
-            return (
-        <div className='article-container'>
-            <ul className="article_list">
-                 {articles.map((article) => (
-                    <Link className='link' key={article.article_id} to ={`/articles/${article.article_id}`}><li className="article_card" key={article.article_id}>
-                        <p>Votes:{article.votes}</p>
-                        <h2 >{article.title}</h2>
-                        <p>by {article.author}</p>
-                        <p>Comments:{article.comment_count}</p>
-                    </li>
-                    </Link>
-                ))}
-              
-            </ul>
-        </div>
+    useEffect(() => {
+        setIsLoading(true);
+        fetchData(`/articles?${searchParams.toString()}&sort_by=${sortBy}&order=${sortOrder}`).then(({articles})=>{
```

Comment:
It's OK to update the url , but axios has its own way to deal with it by taking a query object as the second argument for .get method. It would look neater than trying to updating the url on your frontend.
e.g. `axios.get( "articles" , { params: { topic: topic, sort_by: sortBy, order: order, ...  } })`